### ISSUE TITLE
controller in excmd: Error: text is undefined

### ISSUE THREAD

**Issue by theurgi** [tridactyl/tridactyl#3418]
- Tridactyl version: 1.21.0pre5325-812bb058
- Firefox version: Mozilla Firefox 86.0
- Operating system: macOS 10.15.7 
- Native messenger is correctly installed, version 0.2.5 
- `~/Library/ApplicationSupport/Firefox` was deleted so settings are default
- `toolkit.legacyUserProfileCustomizations.stylesheets = true`

---

In general I can't `: set value` where `value` is any of the available values displayed after `: set`.

For example, the following will not update the config:

<img width="135" alt="image" src="https://user-images.githubusercontent.com/58859663/109406106-6853ee00-7944-11eb-9382-5816dadf84e2.png">

`: get theme` still returns `# theme = default`

Specifically I'm trying to `: set profiledir` so that I can `: guiset gui none`. Also, `: guiset gui none` returns:

```
# controller in excmd: Error: text is undefined
```

**Comment by bovine3dom** [comment]
Hi, thanks for reporting this.

Our set messages are slightly misleadinf

**Comment by theurgi** [comment]
I see. I was using tab enter thinking I was auto completing a command but was actually getting a read value returned.

I've successfully set some configurations and my `profiledir` but am still getting `# controller in excmd: Error: text is undefined` on `guiset`: 🤔

I actually don't think I was having this problem when I originally tried the stable build. I'll give that a shot and report back. Thanks

**Comment by bovine3dom** [comment]
What `:native` are you using? It could be a problem with that since we changed it quite a lot recently.

**Comment by theurgi** [comment]
version 0.2.5

I tried switching to the stable tridactyl release as well as manually removing the `native_main` bin and `trinativeinstall.sh` and then reinstalling. Still no luck.

Can you point me to an earlier release of native messenger and I'll give that a shot?

**Comment by bovine3dom** [comment]
The old messenger can be installed here: https://github.com/tridactyl/tridactyl/blob/master/native/install.sh

It should be able to overwrite userChrome.css.

**Comment by everdred** [comment]
It looks like I might be experiencing the same thing with Firefox 78.7.0esr and Tridactyl 1.21.0 with a fresh Firefox profile. A native install seems to work but then `guiset tabs none`  returns `controller in excmd: Error: text is undefined`.

**Comment by bovine3dom** [comment]
@everdred are you also on OSX?

I can't reproduce this on Arch Linux

**Comment by everdred** [comment]
Nope, this is on Debian (Bullseye).

**Comment by theurgi** [comment]
Running [this version](https://github.com/tridactyl/tridactyl/blob/master/native/install.sh) of the install script worked for me.

I'm now on Native messenger `version 0.1.10` and no longer get an error on `guiset`.

**Comment by everdred** [comment]
@theurgi  Thanks for that link. It worked for me on the first try on a fresh profile (actually a fresh user account) but didn't work on my existing profile (which didn't have a `chrome` subdirectory, much less a `userChrome` file) the first couple of times I tried it. 

But for whatever reason, it works now. Here's the command I used:

`curl -fsSl https://raw.githubusercontent.com/tridactyl/tridactyl/master/native/install.sh  -o /tmp/trinativeinstall.sh && sh /tmp/trinativeinstall.sh 1.21.0`
